### PR TITLE
Update extract_grid.py

### PR DESCRIPTION
**Pull request by bhavesh2109**
VOI was being allocated a huge memory by default and the update extent went outside the whole extent. Tweaked the code a little bit to bring the update extent within whole extent.

### CODE REVIEW

File: mayavi/filters/extract_grid.py [hunk 1]
```diff
@@ -168,12 +168,13 @@ def update_pipeline(self):
 
         fil = self.filter
         self.configure_connection(fil, inputs[0])
-        fil.update_whole_extent()
-        fil.update()
+        self._update_limits()
+        #fil.update_whole_extent()
```
Comment (prabhuramachandran) [inline mayavi/filters/extract_grid.py]: Please remove commented code.

File: mayavi/filters/extract_grid.py [hunk 2]
```diff
@@ -194,6 +195,12 @@ def _update_limits(self):
             extents = self.filter.input.whole_extent
         else:
             extents = self.filter.get_update_extent()
+
+        if extents[0]>extents[1] or extents[2]>extents[3] or extents[4]>extents[5]:
```
Comment (prabhuramachandran) [inline mayavi/filters/extract_grid.py]: Please use PEP8 formatting and no line should exceed, about 78 cols.

**Comment by prabhuramachandran** [comment]
Awesome, thanks for the fix! :)  Please adddress the issues I mention right away and I will merge this.

**Comment by bhavesh2109** [comment]
Done

File: mayavi/filters/extract_grid.py [hunk 3]
```diff
@@ -168,12 +168,12 @@ def update_pipeline(self):
 
         fil = self.filter
         self.configure_connection(fil, inputs[0])
-        fil.update_whole_extent()
-        fil.update()
+        self._update_limits()
         self._set_outputs([fil])
         self._update_limits()
```
Comment (prabhuramachandran) [inline mayavi/filters/extract_grid.py]: Is this second call to update_limits needed now?

Comment (bhavesh2109) [inline mayavi/filters/extract_grid.py]: Fixing it

**Pull request merged**